### PR TITLE
[Tizen/native/wearable] Orientation Detection app

### PR DESCRIPTION
**Pull request by anyj0527**
- Use tizen sensor data and tensorflow-lite model to figure out the orientation of device
- tensor_src_tizensensor element is used to feed the accelerometer sensor data to pipeline

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by myungjoo** [comment]
Great!

File: Tizen.native/OrientationDetection/src/orientationdetection.c [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  float *output;
+  appdata_s *ad = &app_data;
+  int ret;
+
+  G_LOCK (callback_lock);
+
+  ret = ml_tensors_data_get_tensor_data (data, 0, (void **) &output, &data_size);
+  if (ret != ML_ERROR_NONE) {
+    dlog_print (DLOG_ERROR, LOG_TAG, "ml_tensors_data_get_tensor_data failed");
+    return;
```
Comment (jaeyun-jung) [inline Tizen.native/OrientationDetection/src/orientationdetection.c:63]: need G_UNLOCK (callback_lock);

Comment (anyj0527) [reply Tizen.native/OrientationDetection/src/orientationdetection.c]: Thanks! I missed it. I added a goto statement.

File: Tizen.native/OrientationDetection/src/orientationdetection.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+  Evas_Object *conform;
+  Evas_Object *label;
+  Ecore_Pipe *data_output_pipe;
+} appdata_s;
+
+static appdata_s app_data = {0,};
+
+ml_pipeline_h handle;
+ml_pipeline_sink_h sinkhandle;
+gchar * pipeline;
```
Comment (jaeyun-jung) [inline Tizen.native/OrientationDetection/src/orientationdetection.c:41]: let's remove global value (pipeline and tflite_model[] can be moved to init_pipeline())

File: Tizen.native/OrientationDetection/src/orientationdetection.c [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+typedef struct appdata {
+  Evas_Object *win;
+  Evas_Object *conform;
+  Evas_Object *label;
+  Ecore_Pipe *data_output_pipe;
+} appdata_s;
+
+static appdata_s app_data = {0,};
+
+ml_pipeline_h handle;
```
Comment (jaeyun-jung) [inline Tizen.native/OrientationDetection/src/orientationdetection.c:39]: handle and sinkhandle also can be moved to appdata_s

File: Tizen.native/OrientationDetection/src/orientationdetection.c [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+void
+get_output_from_tensor_sink (const ml_tensors_data_h data,
+    const ml_tensors_info_h info, void *user_data)
+{
+  size_t data_size;
+  float *output;
+  appdata_s *ad = &app_data;
+  int ret;
+
+  G_LOCK (callback_lock);
```
Comment (jaeyun-jung) [inline Tizen.native/OrientationDetection/src/orientationdetection.c:59]: Q: why do you use the lock? (now only sink-callback uses this lock)

Comment (anyj0527) [reply Tizen.native/OrientationDetection/src/orientationdetection.c]: Well you're right. I removed the lock

File: Tizen.native/OrientationDetection/src/orientationdetection.c (hunk 2 above)
Comment (anyj0527) [reply Tizen.native/OrientationDetection/src/orientationdetection.c]: I removed the all global variables except the `static app_data`. Please check!

File: Tizen.native/OrientationDetection/src/orientationdetection.c (hunk 3 above)
Comment (anyj0527) [reply Tizen.native/OrientationDetection/src/orientationdetection.c]: Updated!

File: Tizen.native/OrientationDetection/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+3. 6 o'clock is upward.
+4. 9 o'clock is upward.
+
+## Description
+
+* This is a sample application of Tizen Native for wearable device.
+* If you want to run it on your device, Tizen 6.0 or higher is required.
+* About details of NNStreamer, please check [this page](https://docs.tizen.org/application/native/guides/machine-learning/machine-learning-inference).
+
+## Screenshot
```
Comment (gichan-jang) [inline Tizen.native/OrientationDetection/README.md:19]: How about adding a graphical description of the pipeline?

Comment (anyj0527) [reply Tizen.native/OrientationDetection/README.md:25]: I elaborated the README.md please check!

**Pull request merged**